include/xtensor/xassign.hpp

```diff
@@ -299,7 +297,8 @@ namespace xt
         static constexpr bool simd_assign() { return contiguous_layout() && convertible_types() && simd_size() && has_simd_interface<E2>::value; }
         static constexpr bool simd_strided_loop() { return convertible_types() && simd_size() &&
                                                            detail::use_strided_loop<E2>::value &&
-                                                           detail::use_strided_loop<E1>::value; }
+                                                           detail::use_strided_loop<E1>::value &&
+                                                           has_simd_interface<E2>::value; }
```

Comment:
just a comment that it's actually kinda a limitation since even without activating xsimd the strided loop should speed up certain cases.

Reply:
I agree, I will make another pass on this today when I plug the vectorization of pure boolean operations.